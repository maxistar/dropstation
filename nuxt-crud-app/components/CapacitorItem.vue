<template>
  <v-card class="device-item">
    <v-card-text class="device-info">
      {{ device.id }}!
    </v-card-text>
    <v-card-actions class="device-actions">
      <v-btn @click="$emit('edit', device.id)" color="primary">Edit</v-btn>
      <v-btn @click="showConfirmDialog" color="error">Delete</v-btn>
    </v-card-actions>

    <ConfirmationDialog
        :isOpen="dialog"
        title="Confirm Deletion"
        message="Are you sure you want to delete this device?"
        @update:isOpen="dialog = $event"
        @confirm="confirmDelete"
    />
  </v-card>
</template>

<script setup lang="ts">
import { ref } from 'vue';
import { defineProps, defineEmits } from 'vue';
import ConfirmationDialog from '~/components/ConfirmationDialog.vue';

const props = defineProps({
  device: {
    type: Object,
    required: true,
  },
});

const emit = defineEmits(['edit', 'delete']);

const dialog = ref(false);

const showConfirmDialog = () => {
  dialog.value = true;
};

const confirmDelete = () => {
  emit('delete', props.device.id);
};
</script>

<style scoped>
.device-item {
  margin-bottom: 1rem;
}

.device-info {
  flex-grow: 1;
}

.device-actions {
  display: flex;
  justify-content: flex-end;
  gap: 0.5rem;
}
</style>
