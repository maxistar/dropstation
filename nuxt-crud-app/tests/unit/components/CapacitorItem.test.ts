import { mockNuxtImport, mountSuspended } from '@nuxt/test-utils/runtime'
import { it, expect, describe } from 'vitest'
import CapacitorItem from '../../components/CapacitorItem.vue'

describe('AppNumber', () => {
    it('can mount the component', async () => {  
        const component = await mountSuspended(CapacitorItem,  {props: {device: {id:'1'}}})
        expect(component.html()).toContain('v-btn');
    })
})
