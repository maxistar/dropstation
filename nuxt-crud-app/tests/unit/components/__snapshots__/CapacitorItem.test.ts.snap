// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppNumber > match snapshot 1`] = `
"<div data-v-3c7a1c45="" class="v-card v-theme--light v-card--density-default v-card--variant-elevated device-item">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background" style="opacity: NaN;"></div>
      <div class="v-progress-linear__buffer" style="opacity: NaN; width: 0%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div data-v-3c7a1c45="" class="v-card-text device-info">1! </div>
  <div data-v-3c7a1c45="" class="v-card-actions device-actions"><button data-v-3c7a1c45="" type="button" class="v-btn v-btn--slim v-theme--light text-primary v-btn--density-default v-btn--size-default v-btn--variant-text"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator="">Edit</span>
      <!---->
      <!---->
    </button><button data-v-3c7a1c45="" type="button" class="v-btn v-btn--slim v-theme--light text-error v-btn--density-default v-btn--size-default v-btn--variant-text"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
      <!----><span class="v-btn__content" data-no-activator="">Delete</span>
      <!---->
      <!---->
    </button></div>
  <!---->
  <!---->
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;
